src/drivers/alertmsg_can_drv.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Race Capture Firmware
+ *
+ * Copyright (C) 2016 Autosport Labs
+ *
+ * This file is part of the Race Capture firmware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "alertmsg_can_drv.h"
+#include "CAN.h"
+#include "printk.h"
+#include <string.h>
+#include "stdutil.h"
+
+#define _LOG_PFX "[AlertMsgCAN] "
+
+#define DEFAULT_CAN_TIMEOUT   100
+#define ALERTMSG_CAN_BUS 1
+#define ALERTMSG_CAN_BASE_ADDRESS 0xA100
+#define ALERTMSG_MAX_PRIORITY 2
+
+void alertmsg_can_send_message(const struct alertmessage *alertmsg)
+{
+        /* Filter the message  and zero pad */
+        MIN(8, strlen(alertmsg->message));
+        char message_text[17] = {0};
+        strncpy(message_text, alertmsg->message, 16);
+
```

Comment:
Is there a reason, that you copy this here?  It's just copied again at line 52/61.

Reply:
That's true; I was trying to be pedantic and separating the alertmsg->message from the CAN bus data, guaranteeing 0 terminated string.

Reply:
OK.